peer_review_data/main.py

```diff
@@ -99,103 +108,146 @@ def saveAssessmentsAndComments(
+            problemType = 'Assessor'
+            problemObjectId = canvasAssessment.assessorId
+            problemException = e
+        finally:
+            if problemType is not None:
+                LOGGER.warning(f'Error saving Assessment '
+                               f'({canvasAssessment.id}): '
+                               f'{problemType} ({problemObjectId}); '
+                               f'{problemException}')
+                LOGGER.debug('Assessment data: ' + json.dumps(
+                    dictSkipKeys(canvasAssessment, ['_requester']),
+                    indent=2, default=str))
+                continue
+
+        canvasComment: CanvasComment
+        for canvasComment in [CanvasComment(c) for c in
+                              canvasAssessment.comments]:
+            try:
+                comment: models.Comment = \
+                    models.Comment.fromCanvasCommentAndAssessment(
+                        canvasComment, assessment)
+                LOGGER.debug(f'Saving {comment}…')
+                comment.save()
+            except TypeError as e:
+                LOGGER.warning(f'Error saving {comment}: {e}')
+
+
+def processCourseAssignments(canvasCourse: CanvasCourse):
+    courseSaved = False
 
-        if assessment:
-            canvasComment: CanvasComment
-            for canvasComment in [CanvasComment(c) for c in
-                                  canvasAssessment.comments]:
-                try:
-                    comment: models.Comment = \
-                        models.Comment.fromCanvasCommentAndAssessment(
-                            canvasComment, assessment)
-                    LOGGER.debug(f'Saving {comment}…')
-                    comment.save()
-                except Exception as e:
-                    LOGGER.warning(f'Error saving {comment}: {e}')
+    canvasAssignments: List[CanvasAssignment] = \
+        canvasCourse.get_assignments()
 
+    canvasAssignment: CanvasAssignment
+    # Iterate over assignments which have peer reviews
+    for canvasAssignment in filter(
+            lambda a: a.peer_reviews is True, canvasAssignments):
+        LOGGER.debug(f'Found peer reviewed assignment '
+                     f'({canvasAssignment.id}): '
+                     f'"{canvasAssignment.name}"')
 
-def main() -> None:
-    timeStart: datetime = datetime.now(tz=utc)
-    LOGGER.info(f'Start time: {timeStart.isoformat(timespec="milliseconds")}')
+        if 'rubric_settings' not in dir(canvasAssignment):
+            LOGGER.debug(f'Skipping Assignment ({canvasAssignment.id}): '
+                         'No rubric.')
+            continue
 
-    canvasCourse: CanvasCourse = canvas.get_course(COURSE_ID)
-    LOGGER.info(f'Found course ({canvasCourse.id}): "{canvasCourse.name}"')
-    courseSaved = False
-    course: models.Course | None = None
-    user: models.User | None = None
+        assignmentRubricId: int = canvasAssignment.rubric_settings.get(
```

Comment:
I thought that if `rubric_settings` was available, it would always have an `id`.  But, I guess strictly speaking, it could be null, as the signature for `get()` says.

Reply:
Resolved in 2f0dc10.